rust/connlib/snownet/src/backoff.rs

```diff
@@ -1,31 +1,51 @@
 use std::time::{Duration, Instant};
 
-pub type ExponentialBackoff = backoff::exponential::ExponentialBackoff<ManualClock>;
+const MULTIPLIER: f32 = 1.5;
+const MAX_ELAPSED_TIME: Duration = Duration::from_secs(8);
 
 #[derive(Debug)]
-pub struct ManualClock {
-    pub now: Instant,
+pub struct ExponentialBackoff {
+    start_time: Instant,
+    next_trigger: Instant,
+    interval: Duration,
 }
 
-impl backoff::Clock for ManualClock {
-    fn now(&self) -> Instant {
-        self.now
+impl ExponentialBackoff {
+    pub(crate) fn handle_timeout(&mut self, now: Instant) {
+        if self.is_expired(now) {
+            return;
+        }
+
+        if now < self.next_trigger {
+            return;
+        }
+
+        self.interval = Duration::from_secs_f32(self.interval.as_secs_f32() * MULTIPLIER);
+        self.next_trigger += self.interval;
```

Comment:
Would it be a good idea to keep the jitter? For STUN/TURN there's probably a low likelihood that we'll hit the thundering herd problem, but if we end up using this implementation for PhoenixChannel, it could be an issue.

Reply:
> For STUN/TURN there's probably a low likelihood that we'll hit the thundering herd problem

Given the different geographical locations of users and gateways to the relays, I think it is quite unlikely that this matters here. Plus, clients and gateways will all start up at different times and thus are already naturally offset relative to each other.

Reply:
Ok. Just to double-check, we are still using jitter with the PhoenixChannel implementation though?

When a WebSocket connects, we write to the DB, so there's a decent likelihood of hitting this issue there if the `api` service ever restarts or something.

Reply:
I.e. spreading the connects out over 50-100ms is probably not a large enough window of time, considering we sometimes receive latency alarms in the 4-5s range due to DB latency.

Reply:
> Ok. Just to double-check, we are still using jitter with the PhoenixChannel implementation though?

Yes, it is spread +- 50% of the retry interval.